lib/balance-tx/test/spec/Internal/Cardano/Write/Tx/BalanceSpec.hs

```diff
@@ -2204,24 +2212,24 @@ instance Arbitrary (MixedSign Value) where
     shrink (MixedSign v) = MixedSign <$> shrink v
 
 instance forall era. IsRecentEra era => Arbitrary (PartialTx era) where
-    arbitrary = do
-        tx <- genTxForBalancing
-        extraUTxO <- genExtraUTxO (txInputs tx)
-        let redeemers = []
-        let timelockKeyWitnessCounts = mempty
-        pure PartialTx {tx, extraUTxO, redeemers, timelockKeyWitnessCounts}
-      where
-        genExtraUTxO :: Set TxIn -> Gen (UTxO era)
-        genExtraUTxO = fmap UTxO . genMapFromKeysWith genTxOut
-        txInputs :: Tx era -> Set TxIn
-        txInputs tx = tx ^. bodyTxL . inputsTxBodyL
-    shrink partialTx@PartialTx {tx, extraUTxO} =
-        [ partialTx {extraUTxO = extraUTxO'}
-        | extraUTxO' <- shrinkInputResolution extraUTxO
-        ] <>
-        [ restrictResolution (partialTx {tx = tx'})
-        | tx' <- shrinkTx tx
-        ]
+    arbitrary =
+        mkPartialTx <$> TxWithUTxO.generate genTxForBalancing genTxIn genTxOut
```

Comment:
`50%` with surplus UTxO set seems unnecessary? Why not 100% or 90%?

Reply:
@Anviking With the current `Arbitrary` instance for `PartialTx`, the proportion of generated `PartialTx` objects with surplus UTxO sets is 0%:

```hs
instance forall era. IsRecentEra era => Arbitrary (PartialTx era) where
    arbitrary = do
        tx <- genTxForBalancing
        extraUTxO <- genExtraUTxO (txInputs tx)
        pure PartialTx {tx, extraUTxO, redeemers, timelockKeyWitnessCounts}
      where
        genExtraUTxO :: Set TxIn -> Gen (UTxO era)
        genExtraUTxO = fmap UTxO . genMapFromKeysWith genTxOut
        txInputs :: Tx era -> Set TxIn
        txInputs tx = tx ^. bodyTxL . inputsTxBodyL
```

However, the semantics of `balanceTx` do allow for a `PartialTx` to have surplus UTxOs not referenced by the inputs of the inner `Tx`. While this remains true, I think we should at least cover this case in the test suite.

This PR changes the ratio from 0% to 50%, so that we are now (at least some of the time) testing the case where there is a UTxO surplus within the `PartialTx` UTxO set.

Reply:
Apologies - I meant the opposite 😅 : Why not e.g. 10% (going from 0% to 10%)? Most tests in `BalanceSpec` shouldn't care about the possibility for the extra UTxO set to contain a surplus to what is needed.

Reply:
@Anviking Fair enough!

I've adjusted the frequency distribution in this commit:
403ed7df664637b0041c7dce5bd5c92d1a3ec1e8

Reply:
👍 I believe this brought the success rate from 2% back up to 4% (which btw still was very low 👀 )